### DISCUSSION TITLE
How to connect to/use custom credentials

### DISCUSSION THREAD

**Discussion by mgaudette341** [requarks/wiki#6724] [Help / Questions] [answered]
Hi, 

Looking to use Wiki.js - I have a (custom) system with passwords in a DB (hashed, etc). I would like to have Wiki connect to a self-made API that would check with that DB whether the credentials are good or not.

Is there a provision for credentials to be fetch via a custom API? 

Alternatively, I could also save the users/passwords in the Wiki.js DB as they are created in my custom system if I knew how they are hashed in Wiki.JS. Is that info available?

**Comment by NGPixel** [comment, marked as answer]
You can [write your own auth module](https://docs.requarks.io/dev/authentication) or use a standard like OIDC to achieve that.
